src/components/Card/CardNumberInputBox/CardNumberInputBox.tsx

```diff
@@ -0,0 +1,76 @@
+import { ChangeEvent, useEffect, useRef } from 'react';
+
+import { Box, Input } from '@/components/Common';
+import { CardNumber } from '@/types/card';
+import { CARD_NUMBER } from '@/constants/card';
+
+type CardNumberInputProps = {
+  cardNumber: CardNumber;
+  onChange: (e: ChangeEvent<HTMLInputElement>) => void;
+};
+
+export default function CardNumberInputBox({ cardNumber, onChange }: CardNumberInputProps) {
+  const { num1, num2, num3, num4 } = cardNumber;
+
+  const num1Ref = useRef<HTMLInputElement>(null);
+  const num2Ref = useRef<HTMLInputElement>(null);
+  const num3Ref = useRef<HTMLInputElement>(null);
+  const num4Ref = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    num1Ref.current?.focus();
+  }, []);
+
+  useEffect(() => {
+    num1.length === CARD_NUMBER.MAX_LENGTH && num2Ref.current?.focus();
+  }, [num1]);
+
+  useEffect(() => {
+    num2.length === CARD_NUMBER.MAX_LENGTH && num3Ref.current?.focus();
+  }, [num2]);
+
+  useEffect(() => {
+    num3.length === CARD_NUMBER.MAX_LENGTH && num4Ref.current?.focus();
+  }, [num3]);
+
+  return (
+    <Box title="카드 번호" className="input-container">
+      <Input
+        type="text"
+        name="num1"
+        ref={num1Ref}
+        maxLength={CARD_NUMBER.MAX_LENGTH}
+        value={num1}
+        onChange={onChange}
+        className="input-basic w-25"
+      />
+      <Input
+        type="text"
+        name="num2"
+        ref={num2Ref}
+        maxLength={CARD_NUMBER.MAX_LENGTH}
+        value={num2}
+        onChange={onChange}
+        className="input-basic w-25"
+      />
```

Comment:
```suggestion
      <Input
        type="text"
        name="num2"
        ref={num2Ref}
        maxLength={CARD_NUMBER.MAX_LENGTH}
        value={num2}
        onChange={onChange}
        onMaxLengthFilled={() => num2Ref.current?.focus()}
        className="input-basic w-25"
      />
```

useEffect를 별도의 custom hook으로 분리하는 방법도 있을 것 같고..
아니면 위에 제안드린 것 처럼 컴포넌트에서 maxLength를 알고있으니 입력창이 onChange될 때 value의 길이가 maxLength일 때 함수를 받아서 실행해볼수도 있을 것 같아요

Reply:
추가로.. onMaxLengthFilled에서 event를 넘겨받으면 element.nextElementSibling으로 ref.currnet를 대체할수도 있어보여요

Reply:
두 방법 모두 너무 좋아보입니다! DOM API를 이용하는 element.nextElementSibling의 접근은 
전혀 생각해보지 못했는데 좋은 깨달음을 얻은 것 같습니다!
제안 감사드립니다!